src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceCheckedListBox.cs

```diff
@@ -137,23 +137,23 @@ protected override void OnDrawItem(DrawItemEventArgs e)
 
                         var packageSource = new PackageSource(currentItem.Source, currentItem.Name);
                         packageSource.AllowInsecureConnections = currentItem.AllowInsecureConnections;
-                        var shouldShowHttpWarningIcon = packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections;
-                        Rectangle warningBounds = default;
+                        var shouldShowHttpErrorIcon = packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections;
```

Comment:
Editing the URL or changing from http to https would not affect the value of `allowInsecureConnections` for the specific source. However, changing the source name would completely replace the existing source as a result the value `allowInsecureConnections` would be lost. 

I feel like we should add a way for users to see all the properties of a source without having to open the config file. For `allowInsecureConnections` property, we could do that as follows 
![image](https://github.com/NuGet/NuGet.Client/assets/59111203/c17171b4-cb8c-4871-b26b-616d47768317)


This would allow users to know learn about the sources and update them without having to open the config file. What do you think @NuGet/nuget-client ? CC : @JonDouglas

Reply:
wrote a spec for this here :https://github.com/NuGet/Home/pull/13376

Reply:
I would pop up a dialog with something like "NuGet does not recommend using HTTP sources because blah blah.  Do you want to add this HTTP source and disable protections?" with Yes/No buttons.